f8a_worker/workers/recommender.py

```diff
@@ -434,32 +434,33 @@ class RecommendationTask(BaseTask):
 
     def execute(self, arguments=None):
         arguments = self.parent_task_result('GraphAggregatorTask')
+        recommendations = []
         rs = RelativeSimilarity()
 
-        input_stack = {d["package"]: d["version"] for d in arguments.get("result", [])[0].get("details", [])[0].get("_resolved")}
-        ecosystem = arguments.get("result", [])[0].get("details", [])[0].get("ecosystem")
-
-        # Get Input Stack data
-        input_stack_vectors = GraphDB().get_input_stacks_vectors_from_graph(input_stack, ecosystem)
-        # Fetch all reference stacks if any one component from input is present
-        ref_stacks = GraphDB().get_reference_stacks_from_graph(input_stack.keys())
-
-        if len(ref_stacks) > 0:
-            # Apply jaccard similarity to consider only stacks having 30% interection of component names
-            # We only get one top matching reference stack based on components now
-            # filtered_ref_stacks = rs.filter_package(input_stack, ref_stacks)
-            # Calculate similarity of the filtered stacks
-            similar_stacks_list = rs.find_relative_similarity(input_stack, input_stack_vectors, ref_stacks)
-            similarity_list = self._get_stack_values(similar_stacks_list)
-            result = {"recommendations": {
-                "similar_stacks": similarity_list,
-                "component_level": None,
-                }
-            }
-        else:
-            result = {"recommendations": {"similar_stacks": [], "component_level": None,}}
-
-        return result
+        for result in arguments['result']:
+            input_stack = {d["package"]: d["version"] for d in result.get("details", [])[0].get("_resolved")}
+            ecosystem = result.get("details", [])[0].get("ecosystem")
```

Comment:
If there was no `'details'` in `result` you would see `IndexError: list index out of range` because you try to access its first ([0]) member.

If you change it to
```python
ecosystem = result["details"][0].get("ecosystem")
```
you would see `KeyError: 'details'`, which is more self-explanatory.